apps/beeswax/src/beeswax/server/hive_server2_lib.py

```diff
@@ -682,7 +682,11 @@ def open_session(self, user):
 
     LOG.info('Opening %s thrift session for user %s' % (self.query_server['server_name'], user.username))
 
-    req = TOpenSessionReq(**kwargs)
+    try:
+      req = TOpenSessionReq(**kwargs)
+    except Exception as e:
+      if 'Connection refused' in str(e):
+        reset_ha()
```

Comment:
This one is not doing anything? Remove?

Reply:
This is needed to get connected to the new HS2 server if the current one is down.